### PR TITLE
Mix errors from validation and listener

### LINKED ISSUE #3089: Mix errors from validation and listener?
Just found this code in `Documentexecuter`
```csharp
 context = BuildExecutionContext(options, document, operation, variables, metrics);

                foreach (var listener in options.Listeners)
                {
                    await listener.AfterValidationAsync(context, validationResult) // TODO: remove ExecutionContext or make different type ?
                        .ConfigureAwait(false);
                }

                if (!validationResult.IsValid)
                {
                    return new ExecutionResult
                    {
                        Errors = validationResult.Errors,
                        Perf = metrics.Finish()
                    };
                }

                if (context.Errors.Count > 0)
                {
                    return new ExecutionResult
                    {
                        Errors = context.Errors,
                        Perf = metrics.Finish()
                    };
                }

                executionOccurred = true;
```

Questions:
1. If there were validation errors, then is it worth calling `AfterValidationAsync` at all?
2. If yes then why do we have two independent return blocks for validation errors and context errors? They should be mixed.

### PR DESCRIPTION
**Pull request by Shane32**
Closes #3089 
- #3089

### CODE REVIEW

File: src/GraphQL/Execution/DocumentExecuter.cs [hunk 1]
```diff
@@ -148,9 +148,12 @@ private async Task<ExecutionResult> CoreExecuteAsync(ExecutionOptions options)
 
                 if (!validationResult.IsValid)
```
Comment (gao-artur) [inline src/GraphQL/Execution/DocumentExecuter.cs:149]: Can it be `if (!validationResult.IsValid || context.Errors.Count > 0)` and the lines 161-168 removed?

Comment (Shane32) [reply src/GraphQL/Execution/DocumentExecuter.cs:149]: Was trying to reduce allocations, but i can still simplify it some

Comment (Shane32) [reply src/GraphQL/Execution/DocumentExecuter.cs]: Simplified; harder to understand now but less code

**Review by gao-artur — APPROVED** [review]

File: src/GraphQL/Execution/DocumentExecuter.cs [hunk 2]
```diff
@@ -146,20 +146,14 @@ private async Task<ExecutionResult> CoreExecuteAsync(ExecutionOptions options)
                         .ConfigureAwait(false);
                 }
 
-                if (!validationResult.IsValid)
+                if (!validationResult.IsValid || context.Errors.Count > 0)
                 {
+                    var errors = !validationResult.IsValid ? validationResult.Errors : context.Errors;
+                    if (!validationResult.IsValid && context.Errors.Count > 0)
+                        errors.AddRange(context.Errors);
```
[the remark this answers is not in the document]
Comment (Shane32) [reply src/GraphQL/Execution/DocumentExecuter.cs:153]: Working with @sungam3r we always tried real hard to minimize allocations.  Perhaps overly so.  But it makes for a "better" codebase - depending on your viewpoint.

[the remark this answers is not in the document]
Comment (Shane32) [reply src/GraphQL/Execution/DocumentExecuter.cs:153]: This is an error path, not a hot path, so it really is a little overkill.

**Pull request merged**